python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -58,10 +58,25 @@ def register_launch_commands(self):
                     shotgun_data = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_data")
                     # download thumbnail from shotgun
                     self._tk_app.log_debug("Download app icon...")
-                    local_thumb_path = shotgun_data.ShotgunDataRetriever.download_thumbnail(
-                        app_icon_url,
-                        self._tk_app
-                    )
+                    try:
+                        # Download the thumbnail source file from Shotgun, which preserves
+                        # transparency and alpha values.
+                        local_thumb_path = shotgun_data.ShotgunDataRetriever.download_thumbnail_source(
+                            sw_entity["type"], sw_entity["id"], self._tk_app
+                        )
+
+                    except (TankError, AttributeError):
+                        # An old version of tk-framework-shotgunutils or tk-core is
+                        # likely in use. Try ShotgunDataRetriever.download_thumbnail() instead.
+                        self._tk_app.log_warning(
```

Comment:
Maybe the message could say something about what the end result is, i.e. having icons with no transparency. This would help the user understand why upgrading is important.